src/tests/merkle.test.ts

```diff
@@ -13,25 +13,14 @@ describe('validateWitness checks', () => {
     expect(validateWitness(witnessProof, invalidRoot, 2)).toBe(false);
   });
 
-  test('it returns true  for a valid proof and root', () => {
-    const validHashes = ['a', 'b', 'c'].map(sha3_256);
+  test('it returns true for a valid proof and root', () => {
+    const depth = 10;
+    const validHashes = _.range(Math.pow(2, depth)).map(i => sha3_256(i.toString()));
     const validRoot = generateRoot(validHashes);
-    const witnessProof = generateWitness(validHashes, 2);
-
-    expect(validateWitness(witnessProof, validRoot, 2)).toBe(true);
-  });
-});
-
-describe('Proof to index checks', () => {
-  test('4 level tree', () => {
-    const tree = new MerkleTree({hashType: 'SHA3-256'});
-    tree.addLeaves(['0', '1', '2', '3', '4', '5', '6', '7'].map(sha3_256), false);
-    tree.makeTree();
 
-    for (const index of _.range(7)) {
-      const proof = tree.getProof(index);
-      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-      expect(proofToIndex(proof!)).toEqual(index);
+    for (let index = 0; index < depth; index++) {
```

Comment:
Why is the range for this loop `[0..depth-1]`? What about checking all of the leaves instead?

Reply:
89abca1